src/editors/playgroundController.ts

```diff
@@ -492,33 +492,33 @@ export default class PlaygroundController {
     }
 
     try {
-      const progressResult = await vscode.window.withProgress(
+      return await vscode.window.withProgress(
         {
           location: ProgressLocation.Notification,
           title: 'Running MongoDB playground...',
           cancellable: true,
         },
-        async (progress, token) => {
-          token.onCancellationRequested(() => {
-            // If a user clicked the cancel button terminate all playground scripts.
-            this._languageServerController.cancelAll();
-
-            return { result: undefined };
+        async (progress, token): Promise<ShellEvaluateResult> => {
+          // eslint-disable-next-line @typescript-eslint/no-misused-promises
+          return new Promise(async (resolve, reject) => {
```

Comment:
We probably don't want to ignore the [no-misused-promises lint rule](https://typescript-eslint.io/rules/no-misused-promises/). Is it possible to avoid this Promise syntax? I find the resolve/reject syntax can lead to a slippery slope where you start having uncaught exceptions. Same with the one we're adding below in `participantController`.

If possible I think this function should error when it's interrupted, similar to how a `fetch` request will error if the abort signal is aborted.  The consumers of it can then see if the error is an abort error.

Reply:
Totally! Will update 👍